tests/datetime/test_datetime.py

```diff
@@ -27,50 +27,59 @@
 from hypothesis import find, given, assume, settings
 from hypothesis.errors import InvalidArgument
 from tests.common.debug import minimal
+from tests.common.utils import checks_deprecated_behaviour
 from hypothesis.strategytests import strategy_test_suite
 from hypothesis.extra.datetime import datetimes
 from hypothesis.internal.compat import hrange
 
-TestStandardDescriptorFeatures1 = strategy_test_suite(datetimes())
-TestStandardDescriptorFeatures2 = strategy_test_suite(
-    datetimes(allow_naive=False))
-TestStandardDescriptorFeatures3 = strategy_test_suite(
-    datetimes(timezones=[]),
-)
+TestStandardDescriptorFeatures1 = checks_deprecated_behaviour(
+    strategy_test_suite)(datetimes())
+TestStandardDescriptorFeatures2 = checks_deprecated_behaviour(
+    strategy_test_suite)(datetimes(allow_naive=False))
+TestStandardDescriptorFeatures3 = checks_deprecated_behaviour(
+    strategy_test_suite)(datetimes(timezones=[]))
 
 
+@checks_deprecated_behaviour
```

Comment:
If we merge this first, it would also be useful in #580 for a few places where you've put `settings(strict=False)`.